packages/react-native-reanimated/Common/cpp/reanimated/LayoutAnimations/LayoutAnimationsUtils.h

```diff
@@ -53,6 +84,111 @@ struct Snapshot {
   }
 };
 
+inline Float floatFromYogaOptionalFloat(yoga::FloatOptional value) {
+  if (value.isUndefined() || std::isnan(value.unwrap())) {
+    return 0;
+  }
+  return (Float)value.unwrap();
+}
+
+inline Float floatFromYogaBorderWidthValue(const yoga::Style::Length &length) {
+  if (length.unit() == yoga::Unit::Point) {
+    return floatFromYogaOptionalFloat(length.value());
+  }
+  // types percent, and auto are not allowed for border width
+  return 0;
+}
+
+static const int numberOfStringProperties = 2;
+static const char *stringPropertiesNames[2] = {
+    "BackgroundColor",
+    "ShadowColor"};
+
+static const int numberOfNumericProperties = 9;
+static const char *numericPropertiesNames[9] = {
+    "Opacity", // double
+
+    "BorderTopLeftRadius", // int
+    "BorderTopRightRadius",
+    "BorderBottomLeftRadius",
+    "BorderBottomRightRadius",
+
+    "BorderLeftWidth", // float
+    "BorderRightWidth",
+    "BorderTopWidth",
+    "BorderBottomWidth"};
+
+struct StyleSnapshot {
+  std::array<double, 9> numericPropertiesValues;
+  std::array<std::string, 2> stringPropertiesValues;
+
+  StyleSnapshot(
```

Comment:
Is there a way to create a snapshot lazily only with values that have been changed in some way during component lifecycle?

Reply:
I am afraid theres no

Reply:
However, we want the transition to work for set of properties provided by the user, so we still have some space for optimisation here.